src/Symfony/Component/Translation/MessageCatalogue.php

```diff
@@ -232,7 +232,7 @@ public function getResources()
      */
     public function addResource(ResourceInterface $resource)
     {
-        $this->resources[$resource->__toString()] = $resource;
+        $this->resources[(string) $resource] = $resource;
```

Comment:
Good argument against. What is generally agreed approach for components? 

I asking because it's something like just 20 places where __toString is used.